### DISCUSSION TITLE
Scheduler Service is not picking up the schdule

### DISCUSSION THREAD

**Discussion by MubarakPatel** [quartznet/quartznet#1172] [Q&A]
**Describe the bug**
We have use the version 3.0.7 in our dot net core application 
We have created a our windows scheduler service and use the Quartz in it After certain time or in access of application Scheduler Service did not run the upcomings schedules  after the windows service restart everything started working.

After
**Version used**
3.07

**To Reproduce**
Need to keep the scheduler service inactive, after the windows service restart everything started working.

 public async Task Execute(IJobExecutionContext context)
        {
            try
            {
                JobDataMap dataMap = context.JobDetail.JobDataMap;

                string jobId = dataMap.GetString("trubotJobId");
                string command = dataMap.GetString("command") ?? "";
                if (dataMap.GetString("CreatedBy") != null)
                {
                    UserCode = dataMap.GetString("CreatedBy");

                }
                        await ExecuteBotRunner(context).ConfigureAwait(false);
            }
            catch (Exception ex)
            {
                logger.LogError(ex, $"{DateTime.UtcNow} - BotRunnerJob.Execute() - Exception: {ex.Message}", UserCode);
            }
        }
        private async Task ExecuteBotRunner(IJobExecutionContext context)
        {
            JobDataMap dataMap = context.JobDetail.JobDataMap;
            string jobId = dataMap.GetString("trubotJobId");
            string scheduleId = dataMap.GetString("trubotScheduleId");
            string consoleApiUrlRunJob = dataMap.GetString("consoleApiUrlRunJob");
            if (!string.IsNullOrEmpty(consoleApiUrlRunJob))
            {
                
                HttpResponseMessage response = await new HttpClient().
                    PostAsJsonAsync(consoleApiUrlRunJob, new { jobId = jobId, scheduleId = scheduleId }).ConfigureAwait(false);

                await response.Content.ReadAsAsync<string>().ConfigureAwait(false);
            }
        }
**Expected behavior**

**Additional context**

**Comment by lahma** [comment]
* Did you get any error logs? 
* Do you have any logic in your constructor? Constructor throwing exception will prevent job from running
* Are you using persistence? Not being able to connect to database will fail Quartz's startup
* Also check out the new integration libraries that allow to use hosted services, some bug fixes available if you upgrade to latest (not related to this probably though)

**Reply by MubarakPatel** [reply]
@lahma I havent capture any error log in it as it running in the background.
Service is already running in the background and it execute few jobs also but after certain time it didnt take the upcoming schedules
yes i and using Sql server database 
What i feel is some Quartz thread may down/inactive.

![image](https://user-images.githubusercontent.com/43024563/114711931-298bc300-9d4d-11eb-857d-7ee61f72c379.png)
![image](https://user-images.githubusercontent.com/43024563/114712015-41634700-9d4d-11eb-8162-be19ab3eaac8.png)

**Reply by lahma** [reply]
Might want to check [release notes](https://github.com/quartznet/quartznet/releases) since your version if anything seems suitable to your situation.
